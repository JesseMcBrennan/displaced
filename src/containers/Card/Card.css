.moreInfo {
  width: 10%;
  position: center;
}

.Card {
  width: 80rem;
  display: flex;
  justify-content: space-around;
  margin: 2rem 1rem 0 1rem;
  cursor: pointer;
  filter: brightness(95%);
  opacity: 0.95;
  height: 18rem;
  box-shadow: 0 0 80px #000000;
  border-radius: 8px;
  background-color: rgba(255,255,255);
  font-family: 'Open Sans', sans-serif;
  padding-top: 0.5em;
  color: rgba(68,68,68);
}

h2 {
  font-size: 0.5em;
}

a {
  font-size: 0.5em;
}

.iframe-section {
  height: 250px;
  width: 600px;
}

.iframe-section {
  display: flex;
  flex-direction: column;
}

.inlineFrameExample {
  background-color: rgba(255,255,255);
}

.submit-message {
  height: 9em;
}

.text-container {
  text-align: left;
  height: 100%;
  line-height: .5;
  padding-top: 0.4em;
}

