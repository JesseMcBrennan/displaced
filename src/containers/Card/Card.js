import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter, NavLink } from 'react-router-dom';
import { membersFetch, membersMoreInfo } from '../../utils/fetchCalls.js';
import './Card.css'

export class Card extends Component {
  constructor() {
    super()
    this.state = ({
      clicked: false
    })
  }

  handleSubmit = (e) => {
  const url = this.props.member.member_url
  console.log('hi')
  membersMoreInfo(url)
}

  render() {
  const { member } = this.props
  return (
    <div className='Card'>
      <div className='text-container'>
      <h2>{member.role} {member.name}</h2>
      <h2>Party: {member.party}</h2>
      </div>
      <iframe id="inlineFrameExample"
          width="600"
          height="200"
          src="https://www.nytimes.com/search?query=Merkley%252C+Jeff+">
      </iframe>
    </div>
  )
  }
}

export const mapStateToProps = state => ({
  members: state.searchResults
})

export default connect(mapStateToProps)(Card);



