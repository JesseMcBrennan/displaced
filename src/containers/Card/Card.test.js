import { Card } from './Card';
import React from 'react';
import { shallow } from 'enzyme';


describe('Card', () => {
  let wrapper;
  let member = {member:'steve'};
  let event;

  beforeEach(() => {
    event = event = {
      preventDefault: jest.fn(),
      target: {
        value: 'hello',
        name: 'name'
      }
    };
    wrapper = shallow(
      <Card member={member} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
