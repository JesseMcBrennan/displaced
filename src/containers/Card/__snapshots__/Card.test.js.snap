// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    member={
        Object {
            "member": "steve",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="text-container"
>
          <h2>
                     
                     
          </h2>
          <h2>
                    Party: 
          </h2>
          <h2>
                    Voting Percentage: 
                    %
          </h2>
          <h2>
                    Office: 
                     
          </h2>
          <h2>
                    Phone Number: 
                     
          </h2>
          <a
                    href={undefined}
          />
</div>,
        <div
          className="iframe-section"
>
          <iframe
                    id="card-iframe"
                    src={undefined}
                    title="Inline Frame Example"
          />
          <button
                    className="submit-message"
                    onClick={[Function]}
          >
                    Write your Representative
          </button>
</div>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
               
               
</h2>,
            <h2>
              Party: 
</h2>,
            <h2>
              Voting Percentage: 
              %
</h2>,
            <h2>
              Office: 
               
</h2>,
            <h2>
              Phone Number: 
               
</h2>,
            <a
              href={undefined}
/>,
          ],
          "className": "text-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                " ",
                undefined,
                " ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " ",
              undefined,
              " ",
              undefined,
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Party: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Party: ",
              undefined,
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Voting Percentage: ",
                undefined,
                "%",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Voting Percentage: ",
              undefined,
              "%",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Office: ",
                undefined,
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Office: ",
              undefined,
              " ",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Phone Number: ",
                undefined,
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Phone Number: ",
              undefined,
              " ",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "href": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <iframe
              id="card-iframe"
              src={undefined}
              title="Inline Frame Example"
/>,
            <button
              className="submit-message"
              onClick={[Function]}
>
              Write your Representative
</button>,
          ],
          "className": "iframe-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "card-iframe",
              "src": undefined,
              "title": "Inline Frame Example",
            },
            "ref": null,
            "rendered": null,
            "type": "iframe",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Write your Representative",
              "className": "submit-message",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Write your Representative",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="text-container"
>
            <h2>
                         
                         
            </h2>
            <h2>
                        Party: 
            </h2>
            <h2>
                        Voting Percentage: 
                        %
            </h2>
            <h2>
                        Office: 
                         
            </h2>
            <h2>
                        Phone Number: 
                         
            </h2>
            <a
                        href={undefined}
            />
</div>,
          <div
            className="iframe-section"
>
            <iframe
                        id="card-iframe"
                        src={undefined}
                        title="Inline Frame Example"
            />
            <button
                        className="submit-message"
                        onClick={[Function]}
            >
                        Write your Representative
            </button>
</div>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                 
                 
</h2>,
              <h2>
                Party: 
</h2>,
              <h2>
                Voting Percentage: 
                %
</h2>,
              <h2>
                Office: 
                 
</h2>,
              <h2>
                Phone Number: 
                 
</h2>,
              <a
                href={undefined}
/>,
            ],
            "className": "text-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  " ",
                  undefined,
                  " ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " ",
                undefined,
                " ",
                undefined,
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Party: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Party: ",
                undefined,
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Voting Percentage: ",
                  undefined,
                  "%",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Voting Percentage: ",
                undefined,
                "%",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Office: ",
                  undefined,
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Office: ",
                undefined,
                " ",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Phone Number: ",
                  undefined,
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Phone Number: ",
                undefined,
                " ",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "href": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <iframe
                id="card-iframe"
                src={undefined}
                title="Inline Frame Example"
/>,
              <button
                className="submit-message"
                onClick={[Function]}
>
                Write your Representative
</button>,
            ],
            "className": "iframe-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "card-iframe",
                "src": undefined,
                "title": "Inline Frame Example",
              },
              "ref": null,
              "rendered": null,
              "type": "iframe",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Write your Representative",
                "className": "submit-message",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Write your Representative",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
